### ISSUE TITLE
Autopilot Shuttle Rework

### ISSUE THREAD

**Issue by Akbadain29** [unitystation/unitystation#5566]
![image](https://user-images.githubusercontent.com/23048385/98485360-a0360900-2261-11eb-95d2-90a3e465337e.png)

### Description 
Cargo and Evac shuttles have a habit of getting stuck, smashing into asteroids/other shuttles or becoming inoperable with their current set up. This could be addressed by allowing a shuttle (when it is near the station) to first orientate itself correctly a safe distance away before guiding itself using Retro Control System (RCS). The variable names as such should be:
- Arrival point – where the shuttle uses its engines to travel to a particular point after orientating it.
- Docking point – where the shuttle is given an orientation (which it switches to if it isn’t already orientated) and then uses RCS to travel to a particular point
- Warp point - where the shuttle teleports from one co-ordinate to another, via a given z-level for bluespace transit.
- Altitude point - where the shuttle goes to a different z-level scene through a different z-level co-ordinate.

All the above points can be added to a list in the editor to allow a mapper consistent control over a shuttle's path. Currently each special shuttle has it's own script to control just the engines. See Assets/Scripts/Shuttle.

These points are planned out and stored in a dynamic list, which changes if there is an obstacle in the shuttle's path. This should also fix up the autopilot button, which just saves co-ordinates and an orientation for the shuttle to fly to. The autopilot list should always save the shuttle's starting co-ordinates and orientation, but further options need to be manually saved. This shouldn't be too hard as shuttles have a radar on the shuttle console and are able to see other asteroids and shuttles. It'll make piloting the shuttles easier too.

**Comment by Macoron** [comment]
Maybe add warp drive for such shuttles? They could teleport to different locations through hyperspace. 
Hyperspace can be different z-level or just very far location.

**Comment by corp-0** [comment]
Yeah, we can already teleport the shuttles. What I think would be the best is to make them move a little and then play the hyperjump animation and teleport them away. Then you have no need to make intrincated path finding and avoid collisions with dynamic objects

**Comment by PetMudstone** [comment]
Could have certain shuttles hyperjump without moving. Would be nice for putting shuttles inside the station (such as with Pubby's shuttle) instead of being limited to the perimeter. A nice option for mappers.

**Comment by AtnerNT** [comment]
Bod added Shuttle pathfinding in shuttle rework

**Comment by MaxIsJoe** [comment]
I'll be blunt
the current pathfinding system for shuttles is functional, but the "finding" in pathfinding is questionable. Shuttles can still blindly ram into stuff, and cargo shuttles keep losing their orders.
